hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/snapshot/OzoneManagerSnapshotProvider.java

```diff
@@ -135,6 +135,12 @@ public DBCheckpoint getOzoneManagerDBSnapshot(String leaderOMNodeID)
 
       try (InputStream inputStream = httpURLConnection.getInputStream()) {
         FileUtils.copyInputStreamToFile(inputStream, targetFile);
+      } catch (IOException ex) {
+        boolean deleted = FileUtils.deleteQuietly(targetFile);
+        LOG.error("OM snapshot {} cannot be downloaded.", targetFile, ex);
+        if (!deleted) {
+          LOG.error("OM snapshot {} cannot be deleted", targetFile);
```

Comment:
The log messages "OM snapshot cannot be deleted" could be confusing. Can we change the 2nd message to indicate that the "failed snapshot download" could not be deleted?

Reply:
> The log messages "OM snapshot cannot be deleted" could be confusing. Can we change the 2nd message to indicate that the "failed snapshot download" could not be deleted?

Make sense. Thx for the suggestion. I've updated PR. @hanishakoneru